tardis/rest/service.py

```diff
@@ -0,0 +1,50 @@
+from .app.security import UserCredentials
+from cobald.daemon import service
+from cobald.daemon.plugins import yaml_tag
+
+from uvicorn.config import Config
+from uvicorn.server import Server
+
+from functools import lru_cache
+from typing import List
+import asyncio
+
+
+@service(flavour=asyncio)
+@yaml_tag(eager=True)
+class RestService(object):
+    def __init__(
+        self,
+        secret_key: str,
+        algorithm: str = "HS256",
+        users: List = None,
+        **fast_api_args,
+    ):
+        self._algorithm = algorithm
+        self._secret_key = secret_key
+        self._users = users or []
+
+        # necessary to avoid that the TARDIS' logger configuration is overwritten!
+        if "log_config" not in fast_api_args:
+            fast_api_args["log_config"] = None
+        self._config = Config("tardis.rest.app.main:app", **fast_api_args)
+
+    @property
+    @lru_cache(maxsize=16)
+    def algorithm(self) -> str:
+        return self._algorithm
+
+    @property
+    @lru_cache(maxsize=16)
+    def secret_key(self) -> str:
+        return self._secret_key
+
+    @lru_cache(maxsize=16)
+    def get_user(self, user_name: str) -> [None, UserCredentials]:
```

Comment:
Do you mind checking the return type? I would expect `Optional[UserCredentials]` from just reading the code below.

Reply:
Thanks, I have changed it to `Optional[UserCredentials]`.